lib/modules/platform/github/index.ts

```diff
@@ -742,8 +742,9 @@ async function ensureBranchSha(branchName: string, sha: string): Promise<void> {
   const refUrl = `/repos/${config.repository}/git/refs/heads/${branchName}`;
```

Comment:
Let's also check for commit existence:

```suggestion
  const commitUrl = `/repos/${config.repository}/git/commits/sha`;
  await githubApi.head(commitUrl, { memCache: false });

  const refUrl = `/repos/${config.repository}/git/refs/heads/${branchName}`;
```

Reply:
I've reproduced 422 by trying to patch existing branch with non-existent commit SHA. This explains why people suffer in both `PATCH` and `POST` branches of execution. Probably we needed to check the commit existence in the first place, when created `ensureBranchSha`.